assets/js/modules/analytics/datastore/accounts.js

```diff
@@ -155,6 +166,35 @@ const baseActions = {
 			}
 
 			yield propertyActions.selectProperty( uaProperty?.id, uaProperty?.internalWebPropertyId ); // eslint-disable-line sitekit/acronym-case
+
+			if ( ! isFeatureEnabled( 'ga4setup' ) ) {
+				return;
+			}
+
+			registry.dispatch( STORE_NAME ).setPrimaryPropertyType( PROPERTY_TYPE_UA );
+
+			yield Data.commonActions.await( registry.dispatch( MODULES_ANALYTICS_4 ).waitForProperties( accountID ) );
+
+			const ga4Properties = registry.select( MODULES_ANALYTICS_4 ).getProperties( accountID );
+			const ga4Property = yield Data.commonActions.await(
+				registry.dispatch( MODULES_ANALYTICS_4 ).matchPropertyByURL(
+					ga4Properties.map( ( property ) => property._id ),
+					registry.select( CORE_SITE ).getReferenceSiteURL(),
+				)
+			);
+
+			yield Data.commonActions.await(
+				registry.dispatch( MODULES_ANALYTICS_4 ).selectProperty( ga4Property?._id || GA4_PROPERTY_CREATE ),
+			);
```

Comment:
Perhaps it would be more appropriate for this to be handled in an action on the GA4 store (e.g. `matchProperty( accountID )` rather than including it in the UA store? Then this could await the result of the single action here which would return the matched property, if any. I think that would be a little cleaner (both here and in the other store since we wouldn't need to use `registry.dispatch` there) and could be used in both places here.

Reply:
It's a great idea! Updated.